### PR TITLE
Add notify error on project creation form

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

File: src/components/other/ProjectAddForm/index.js [hunk 1]
```diff
@@ -58,142 +60,131 @@ const mapDispatchToProps = dispatch => ({
...
+        onSuccess: (response) => {
+            setUserProject({ project: response });
+            setUsergroupProject({ project: response });
+            setUserProfileProject({
+                userId,
+                project: response,
+            });
+            if (onProjectAdd) {
+                onProjectAdd(response.id);
+            }
```
Comment (tnagorra) [inline src/components/other/ProjectAddForm/index.js:107]: Is this correct? Shouldn't onProjectAdd accept `response` then the parent handle setUserProject, setUsergroupProject and setUserProfileProject on the parent itself?

File: src/components/other/ProjectAddForm/index.js [hunk 2]
```diff
@@ -58,142 +60,131 @@ const mapDispatchToProps = dispatch => ({
...
+                message: _ts('components.addProject', 'userProjectCreateSuccess'),
+                duration: notify.duration.MEDIUM,
+            });
+            onModalClose();
+        },
+        onFailure: (error, errorBody) => {
+            setFaramErrors(errorBody?.faramErrors);
+            setPristine(true);
+            notifyOnFailure(_ts('components.addProject', 'addProjectModalLabel'))({ error: errorBody });
+        },
```
Comment (tnagorra) [inline src/components/other/ProjectAddForm/index.js:119]: handle Fatal condition

File: src/components/other/ProjectAddForm/index.js [hunk 3]
```diff
@@ -58,142 +60,131 @@ const mapDispatchToProps = dispatch => ({
...
+
+    const handleFaramChange = useCallback((newFaramValues, newFaramErrors) => {
+        setFaramValues(newFaramValues);
+        setFaramErrors(newFaramErrors);
+        setPristine(false);
+    }, [setFaramValues, setFaramErrors, setPristine]);
+
+    const handleFaramFailure = useCallback((newFaramErrors) => {
+        setFaramErrors(newFaramErrors);
+        setPristine(true);
```
Comment (tnagorra) [inline src/components/other/ProjectAddForm/index.js]: failure shouid not set pristine to true

File: src/components/other/ProjectAddForm/index.js [hunk 4]
```diff
@@ -58,142 +60,131 @@ const mapDispatchToProps = dispatch => ({
...
+    }, [setFaramErrors, setPristine]);
+
+    const handleFaramSuccess = useCallback(() => {
+        createNewProject();
+    }, [createNewProject]);
```
Comment (tnagorra) [inline src/components/other/ProjectAddForm/index.js]: const handleFaramSuccess = createNewProject;

Comment (samshara) [reply src/components/other/ProjectAddForm/index.js]: if it is so then can't we use createNewProject directly as props to Faram.

File: src/components/other/ProjectAddForm/index.js (hunk 1 above)
Comment (AdityaKhatri) [reply src/components/other/ProjectAddForm/index.js:107]: This was done previously and I replicated it

File: src/components/other/ProjectAddForm/index.js (hunk 2 above)
Comment (AdityaKhatri) [reply src/components/other/ProjectAddForm/index.js:119]: I was under the impression that the new request doesn't have fatal

File: src/components/other/ProjectAddForm/index.js (hunk 3 above)
Comment (AdityaKhatri) [reply src/components/other/ProjectAddForm/index.js]: yes it should, why not?

**Review by samshara — APPROVED** [review]

File: src/components/other/ProjectAddForm/index.js (hunk 4 above)
Comment (tnagorra) [reply src/components/other/ProjectAddForm/index.js]: we can yes sir!

File: src/components/other/ProjectAddForm/index.js (hunk 2 above)
Comment (AdityaKhatri) [reply src/components/other/ProjectAddForm/index.js:119]: I've added a note

**Pull request merged**